src/main/python/systemml/defmatrix.py

```diff
@@ -60,24 +67,42 @@ def __init__(self, inputs, dml=None):
         self.inputs = inputs
         self.dml = dml
         self.ID = None
+        self.depth = 1
         for m in self.inputs:
             m.referenced = m.referenced + [ self ]
+            if isinstance(m, matrix) and m.op is not None:
+                self.depth = max(self.depth, m.op.depth + 1)
 
+    MAX_DEPTH = 0
+    
     def _visit(self, execute=True):
         matrix.dml = matrix.dml + self.dml
 
-    # Don't use this method instead use matrix's printAST()
-    def printAST(self, numSpaces):
+    # Don't use this method instead use matrix's print_ast()
+    def print_ast(self, numSpaces):
```

Comment:
Let's rename this to `_print_ast` to indicate that it is a private method not for use by the user.